### PR TITLE
Expanded Contributing Guidelines (fixes: #241)

### PR DESCRIPTION
**Pull request by cemkod**
Modified three.js CONTRIBUTING.md to suit our contribution process.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,6 +1,28 @@
+Before reporting a bug
+---
+1. Search issue tracker for similar issues.
```
Comment (cvan) [inline CONTRIBUTING.md]: can you add a link here?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+2. Try the latest dev branch version of aframe and aframe-core.
```
Comment (cvan) [inline CONTRIBUTING.md]: can you link to https://github.com/aframevr/aframe/tree/dev and https://github.com/aframevr/aframe-core/tree/dev here?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,6 +1,28 @@
...
+
+How to report a bug
+---
+1. Specify the revision number of the aframe.js library where the bug occurred.
+2. If you are using a PC, specify your browser version, operating system, and graphics card. (for example, Chrome 23.0.1271.95, Windows 7, Nvidia Quadro 2000M)
```
Comment (cvan) [inline CONTRIBUTING.md]: I would move the `.` in `graphics card.` to be after the parentheses.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,6 +1,28 @@
...
+3. If you are using a mobile device, specify your make, model, operating system version and browser. (for example, Samsung Galaxy Nexus, Android 4.4.4, Chrome 47.0.2526.83)
```
Comment (cvan) [inline CONTRIBUTING.md]: add a comma before `and` (for consistency with sentence above)

Comment (cvan) [inline CONTRIBUTING.md]: same here with punctuation

**Comment by cemkod** [comment]
Fixed links and punctuation

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,6 +1,28 @@
...
+2. If you are using a PC, specify your browser version, operating system, and graphics card (for example, Chrome 23.0.1271.95, Windows 7, Nvidia Quadro 2000M).
+3. If you are using a mobile device, specify your make, model, operating system version, and browser (for example, Samsung Galaxy Nexus, Android 4.4.4, Chrome 47.0.2526.83).
+3. Describe the problem in detail. Explain what happened, and what you expected would happen.
+4. Provide a small test-case (http://jsfiddle.net). [Here is a fiddle](http://jsfiddle.net/8tvqo8jh/) that uses the dev branch. If a test-case is not possible, provide a link to a live version of your application.
+5. If helpful, include a screenshot. Annotate the screenshot for clarity.
+
+How to contribute to aframevr
+---
+1. Make sure you have a GitHub account.
+2. Fork the repository on GitHub.
```
Comment (cvan) [inline CONTRIBUTING.md]: you could link to https://github.com/aframevr/aframe/fork

**Comment by cemkod** [comment]
Done, anything else? Should we direct people to codepen instead of jsfiddle? Does it matter?

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,6 +1,28 @@
+Before reporting a bug
```
Comment (cvan) [inline CONTRIBUTING.md]: to be consitent with usage elsewhere, I would use the Markdown `#` and `##` syntax in the future

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,6 +1,28 @@
...
+---
+1. Search [issue tracker](https://github.com/aframevr/aframe/issues) for similar issues.
```
Comment (cvan) [inline CONTRIBUTING.md]: "Search the …"

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,6 +1,28 @@
...
+---
+1. Search [issue tracker](https://github.com/aframevr/aframe/issues) for similar issues. 
+2. Try the latest dev branch version of [aframe](https://github.com/aframevr/aframe/tree/dev) and [aframe-core](https://github.com/aframevr/aframe-core/tree/dev).
+
+How to report a bug
+---
+1. Specify the revision number of the aframe.js library where the bug occurred.
+2. If you are using a PC, specify your browser version, operating system, and graphics card (for example, Chrome 23.0.1271.95, Windows 7, Nvidia Quadro 2000M).
+3. If you are using a mobile device, specify your make, model, operating system version, and browser (for example, Samsung Galaxy Nexus, Android 4.4.4, Chrome 47.0.2526.83).
+3. Describe the problem in detail. Explain what happened, and what you expected would happen.
```
Comment (cvan) [inline CONTRIBUTING.md]: > Describe the problem in detail (i.e., what happened and what you expected would happen).

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,6 +1,28 @@
...
+4. Provide a small test-case (http://jsfiddle.net). [Here is a fiddle](http://jsfiddle.net/8tvqo8jh/) that uses the dev branch. If a test-case is not possible, provide a link to a live version of your application.
+5. If helpful, include a screenshot. Annotate the screenshot for clarity.
+
+How to contribute to aframevr
```
Comment (cvan) [inline CONTRIBUTING.md]: `aframevr` -> `aframe`

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,6 +1,28 @@
...
+---
+1. Make sure you have a GitHub account.
```
Comment (cvan) [inline CONTRIBUTING.md]: if you want to get fancy, even https://github.com/join heh

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -1,6 +1,28 @@
...
+1. Search [issue tracker](https://github.com/aframevr/aframe/issues) for similar issues. 
+2. Try the latest dev branch version of [aframe](https://github.com/aframevr/aframe/tree/dev) and [aframe-core](https://github.com/aframevr/aframe-core/tree/dev).
+
+How to report a bug
+---
+1. Specify the revision number of the aframe.js library where the bug occurred.
+2. If you are using a PC, specify your browser version, operating system, and graphics card (for example, Chrome 23.0.1271.95, Windows 7, Nvidia Quadro 2000M).
+3. If you are using a mobile device, specify your make, model, operating system version, and browser (for example, Samsung Galaxy Nexus, Android 4.4.4, Chrome 47.0.2526.83).
+3. Describe the problem in detail. Explain what happened, and what you expected would happen.
+4. Provide a small test-case (http://jsfiddle.net). [Here is a fiddle](http://jsfiddle.net/8tvqo8jh/) that uses the dev branch. If a test-case is not possible, provide a link to a live version of your application.
```
Comment (cvan) [inline CONTRIBUTING.md]: the only concern I have with recommending JS Fiddle is that you can't enter VR from the Edit mode (only in presentation mode). see https://github.com/aframevr/aframe-core/issues/725

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,6 +1,28 @@
...
+5. If helpful, include a screenshot. Annotate the screenshot for clarity.
+
+How to contribute to aframevr
+---
+1. Make sure you have a GitHub account.
+2. [Fork](https://github.com/aframevr/aframe/fork) the repository on GitHub.
+3. Checkout the [dev branch](https://github.com/aframevr/aframe/tree/dev).
+4. Make changes to your clone of the repository.
```
Comment (cvan) [inline CONTRIBUTING.md]: `clone` -> `fork`

**Comment by cvan** [comment]
thanks so much for this; much appreciated!

**Comment by cemkod** [comment]
No problem :metal: 
All notes addresses. You can check again.

**Comment by cemkod** [comment]
I have added some instructions regarding slack.
See: https://github.com/aframevr/aframe/issues/364

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -1,6 +1,29 @@
...
+5. If helpful, include a screenshot. Annotate the screenshot for clarity.
+
+# How to join us on slack
+1. Request an invitation from our [automated system](https://aframevr-slack.herokuapp.com/)
+2. [Join](https://aframevr.slack.com)  the discussion
+
+# How to contribute to aframe
+1. Make sure you have a [GitHub account](https://github.com/join).
+2. [Fork](https://github.com/aframevr/aframe/fork) the repository on GitHub.
+3. Checkout the [dev branch](https://github.com/aframevr/aframe/tree/dev).
```
Comment (ngokevin) [inline CONTRIBUTING.md]: `Check out`

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -1,6 +1,29 @@
...
+4. Make changes to your fork of the repository.
+5. Submit a pull request.
```
Comment (ngokevin) [inline CONTRIBUTING.md]: + to the `dev` branch

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -1,6 +1,29 @@
-Before contributing code, please note:
+# Before reporting a bug
+1. Search the [issue tracker](https://github.com/aframevr/aframe/issues) for similar issues. 
+2. Try the latest dev branch version of [aframe](https://github.com/aframevr/aframe/tree/dev).
```
Comment (ngokevin) [inline CONTRIBUTING.md]: I don't think we need people to check out the dev branch before reporting an issue.

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -1,6 +1,29 @@
...
+
+# How to report a bug
+1. Specify the revision number of the aframe.js library where the bug occurred.
```
Comment (ngokevin) [inline CONTRIBUTING.md]: maybe `Specify the version of the A-Frame where the bug occurred.`

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -1,6 +1,29 @@
...
+2. If you are using a PC, specify your browser version, operating system, and graphics card (for example, Chrome 23.0.1271.95, Windows 7, Nvidia Quadro 2000M).
```
Comment (ngokevin) [inline CONTRIBUTING.md]: For 2/3, maybe just simplify it to `Specify information about your browser and system (e.g., "Firefox Nightly on OS X")`

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -1,6 +1,29 @@
...
+3. If you are using a mobile device, specify your make, model, operating system version, and browser (for example, Samsung Galaxy Nexus, Android 4.4.4, Chrome 47.0.2526.83).
+3. Describe the problem in detail (i.e., what happened and what you expected would happen).
+4. Provide a small test-case (http://codepen.io). [Here is a pen](http://codepen.io/anon/pen/KVWpbb) that uses the dev branch. If a test-case is not possible, provide a link to a live version of your application.
```
Comment (ngokevin) [inline CONTRIBUTING.md]: I think 4/5 should be combined/optional. "If possible, provide a small test case with Codepen, a link to your application, and/or a screenshot. Here is a pen..."

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -1,6 +1,29 @@
...
+5. If helpful, include a screenshot. Annotate the screenshot for clarity.
+
+# How to join us on slack
+1. Request an invitation from our [automated system](https://aframevr-slack.herokuapp.com/)
```
Comment (ngokevin) [inline CONTRIBUTING.md]: `Invite yourself to [our Slack channel]` so that they know they don't have to wait for approval

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -1,6 +1,29 @@
...
+1. Make sure you have a [GitHub account](https://github.com/join).
+2. [Fork](https://github.com/aframevr/aframe/fork) the repository on GitHub.
+3. Checkout the [dev branch](https://github.com/aframevr/aframe/tree/dev).
+4. Make changes to your fork of the repository.
+5. Submit a pull request.
+
+### Before contributing code, please note:
 
 * You agree to license your contributions under the [license](LICENSE).
 * Follow [jshint](http://eslint.org/docs/rules/) [see our rules](.jshintrc).
```
Comment (ngokevin) [inline CONTRIBUTING.md]: We follow https://github.com/Flet/semistandard...but Travis will pick it up for people

File: CONTRIBUTING.md [hunk 21]
```diff
@@ -1,6 +1,29 @@
...
+3. If you are using a mobile device, specify your make, model, operating system version, and browser (for example, Samsung Galaxy Nexus, Android 4.4.4, Chrome 47.0.2526.83).
+3. Describe the problem in detail (i.e., what happened and what you expected would happen).
+4. Provide a small test-case (http://codepen.io). [Here is a pen](http://codepen.io/anon/pen/KVWpbb) that uses the dev branch. If a test-case is not possible, provide a link to a live version of your application.
+5. If helpful, include a screenshot. Annotate the screenshot for clarity.
+
+# How to join us on slack
+1. Request an invitation from our [automated system](https://aframevr-slack.herokuapp.com/)
+2. [Join](https://aframevr.slack.com)  the discussion
+
+# How to contribute to aframe
```
Comment (cvan) [inline CONTRIBUTING.md]: `A-Frame`

File: CONTRIBUTING.md [hunk 22]
```diff
@@ -1,6 +1,29 @@
...
+
+# How to report a bug
+1. Specify the revision number of the aframe.js library where the bug occurred.
+2. If you are using a PC, specify your browser version, operating system, and graphics card (for example, Chrome 23.0.1271.95, Windows 7, Nvidia Quadro 2000M).
+3. If you are using a mobile device, specify your make, model, operating system version, and browser (for example, Samsung Galaxy Nexus, Android 4.4.4, Chrome 47.0.2526.83).
+3. Describe the problem in detail (i.e., what happened and what you expected would happen).
+4. Provide a small test-case (http://codepen.io). [Here is a pen](http://codepen.io/anon/pen/KVWpbb) that uses the dev branch. If a test-case is not possible, provide a link to a live version of your application.
+5. If helpful, include a screenshot. Annotate the screenshot for clarity.
+
+# How to join us on slack
```
Comment (cvan) [inline CONTRIBUTING.md]: `Slack`

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -1,6 +1,29 @@
...
+1. Request an invitation from our [automated system](https://aframevr-slack.herokuapp.com/)
+2. [Join](https://aframevr.slack.com)  the discussion
```
Comment (cvan) [inline CONTRIBUTING.md]: collapse two spaces to one space

File: CONTRIBUTING.md (hunk 20 above)
Comment (cemkod) [inline CONTRIBUTING.md]: @ngokevin: Can you elaborate on this? Should I remove the item about jshint as this is taken care of automatically?

Comment (ngokevin) [inline CONTRIBUTING.md]: Yeah, sorry. I think we can remove the jshint note.

**Comment by ngokevin** [comment]
Thanks for addressing all of our nit-picky comments :)

Can you squash your commits? 

(`git rebase -i HEAD~20` just in case)

File: CONTRIBUTING.md [hunk 24]
```diff
@@ -1,6 +1,26 @@
-Before contributing code, please note:
+# How to report a bug
+1. Search the [issue tracker](https://github.com/aframevr/aframe/issues) for similar issues. 
+2. Specify the version of the A-Frame where the bug occurred.
+3. Specify information about your browser and system (e.g., "Firefox Nightly on OS X")
+4. Describe the problem in detail (i.e., what happened and what you expected would happen).
+5. If possible, provide a small test case with [Codepen](http://codepen.io), a link to your application, and/or a screenshot. [Here](http://codepen.io/anon/pen/KVWpbb) is a sample pen you can fork.
+6. Provide a small test-case (http://codepen.io). [Here is a pen](http://codepen.io/anon/pen/KVWpbb) that uses the dev branch. If a test-case is not possible, provide a link to a live version of your application.
```
Comment (ngokevin) [inline CONTRIBUTING.md]: remove this line

File: CONTRIBUTING.md [hunk 25]
```diff
@@ -1,6 +1,26 @@
-Before contributing code, please note:
+# How to report a bug
+1. Search the [issue tracker](https://github.com/aframevr/aframe/issues) for similar issues. 
+2. Specify the version of the A-Frame where the bug occurred.
```
Comment (ngokevin) [inline CONTRIBUTING.md]: s/the A-Frame/A-Frame

**Comment by cemkod** [comment]
No problem :) I expected many edits when I took on CONTRIBUTING.md
Thanks for putting up with my noob mistakes.

**Comment by ngokevin** [comment]
Thanks!!

**Pull request merged**